src/Chapter2.hs

```diff
@@ -842,7 +856,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate x xs
+  | x < 0     = []
+  | null xs   = []
+  | otherwise = drop x $ take ((length xs) + x) $ cycle xs
```

Comment:
Great!
Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. By using `mod x len` you could optimize this 🚤